### ISSUE TITLE
syslog being called incorrectly

### ISSUE THREAD

**Issue by stylewarning** [mmaul/cl-syslog#5]
The prototype of syslog is

`void syslog(int pri, const char *format_string, ...)`

We should be doing the call equivalent to 

`syslog(user_provided_priority, "%s", user_provided_string)`

**Comment by mmaul** [comment]
I've added initial support for the varargs in logva fn, function sig for syslog is fixed. Please check out branch logva. Also have removed with-foreign-strings from log fn as it wasn't necessary, wonder if it may have had impact to your Memory Fault issues.

**Comment by stylewarning** [comment]
I don't think we should support varargs on the Lisp side. The reason is because the format string is expected to be C-like. I think we should fix the format string to be `"%s"` and just use one single argument after that which is the string that the Lisp user provides.

**Comment by mmaul** [comment]
The printf format is clunky, single string or lisp format style is better.
I did wan't to give you access to va to see if that made any difference
w/memory fault issue and if adding parameters changed anything.

**Comment by stylewarning** [comment]
It doesn't appear to. I think I need to try with CCL or something.

**Comment by stylewarning** [comment]
I fixed this in the to-be-merged PR.
